src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -401,6 +421,37 @@ private void ensureSelectedMedHistIsValid(ListChangeListener.Change<? extends Me
         }
     }
 
+    /**
+     * Ensures {@code selectedAppointment} is a valid appointment in {@code filteredAppointments}.
+     */
+    private void ensureSelectedAppointmentIsValid(ListChangeListener.Change<? extends Appointment> change) {
+        while (change.next()) {
+            if (selectedAppointment.getValue() == null) {
+                // null is always a valid selected patient, so we do not need to check that it is valid anymore.
+                return;
+            }
+
+            boolean wasSelectedAppointmentReplaced =
+                    change.wasReplaced() && change.getAddedSize() == change.getRemovedSize()
```

Comment:
hacky way of comparing the size of the change to check if the appointment is replaced. Perhaps can find a better check by comparing other factors